proto/regen/ecocredit/v1alpha1/events.proto

```diff
@@ -67,3 +67,19 @@ message EventRetire {
   // units is the decimal number of credits that have been retired.
   string units = 3;
 }
+
+// EventCancel is an event emitted when credits are cancelled. When credits are
+// cancelled from multiple batches in the same transaction, a separate event is
+// emitted for each batch_denom. This allows for easier indexing.
+message EventCancel {
+
+  // canceller is the account which has cancelled the credits, which should be
+  // the holder of the credits.
+  string canceller = 1;
+
+  // batch_denom is the unique ID of credit batch.
+  string batch_denom = 2 [ (gogoproto.moretags) = "yaml:\"batch_denom\"" ];
+
+  // units is the decimal number of credits that have been retired.
+  string units = 3;
```

Comment:
Should we say `units` or `amount`?

Reply:
I think I introduced the language `units` but it seems weird to me now

Reply:
Yeah, I agree! I was aiming for consistency here.